codice/ddf
# DDF-5088 Remove empty tag filer on TagsFilterQueryPlugin

#### What does this PR do?
To support legacy metacard without tags, TagsFilterQueryPlugin add an empty metacard-tag filter to the query filter.

(original query) AND (metacard-tag = resource OR metacard-tag is EMPTY)

This injected empty filter invalidate the filter when using TagsFilterDelegate to validate existing of metacard tags

For legacy system with metacard having no tags. Migration can be done by catalog:dumping all the record and catalog:ingest them back in.

The ingest process should tag all metacard as resource and valid or invalid
https://github.com/codice/ddf/blob/master/distribution/ddf-common/src/main/resources/etc/definitions/ddf-catalog-taxonomy-injectors.json#L247


#### Who is reviewing it? 

@codice/core-apis 
@pklinef
@troymohl
@bdeining 

#### How should this be tested?

- turn on solr logging: log:set TRACE ddf.catalog.source.solr
- perform a query from ui
- verify the solr query print out does not contain an metacard-tag empty filter

`OR  (*:* -metacard-tags_txt:[* TO *]) `should **NOT** be in the print out

original query before this PR
`SolrMetacardClientImpl  515 | re-standardframework | Prepared Query:  (  ( metacard-tags_txt_tokenized:"resource" OR  (*:* -metacard-tags_txt:[* TO *]) )  AND  (  (*:* -validation-errors_txt:[* TO *])  AND *:* )  ) `


#### Any background context you want to provide?
#### What are the relevant tickets?
Fixes: https://github.com/codice/ddf/issues/5088
#### Screenshots

#### Checklist:
- [x ] Update / Add Unit Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
